core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -25,32 +25,84 @@ oppia.constant(
-    SkillListObjectFactory, CollectionUpdateService, UndoRedoService,
+    CollectionRightsBackendApiService, ExplorationSummaryBackendApiService,
+    CollectionObjectFactory, SkillListObjectFactory,
+    CollectionValidationService, CollectionUpdateService, UndoRedoService,
     alertsService) {
     $scope.collection = null;
     $scope.collectionId = GLOBALS.collectionId;
     $scope.collectionSkillList = SkillListObjectFactory.create([]);
     $scope.isPublic = GLOBALS.isPublic;
+    $scope.validationIssues = [];
+
+    var _validateCollection = function() {
+      if ($scope.isPublic) {
+        $scope.validationIssues = (
+          CollectionValidationService.findValidationIssuesForPublicCollection(
+            $scope.collection));
+      } else {
+        $scope.validationIssues = (
+          CollectionValidationService.findValidationIssuesForPrivateCollection(
+            $scope.collection));
+      }
+    };
+
+    var _updateCollection = function(newBackendCollectionObject) {
+      $scope.collection = CollectionObjectFactory.create(
+        newBackendCollectionObject);
+      $scope.collectionSkillList.setSkills(newBackendCollectionObject.skills);
+      _validateCollection();
+    };
 
     // Load the collection to be edited.
     WritableCollectionBackendApiService.fetchWritableCollection(
       $scope.collectionId).then(
-        function(collectionBackendObject) {
-          $scope.collection = CollectionObjectFactory.create(
-            collectionBackendObject);
-          $scope.collectionSkillList.setSkills(collectionBackendObject.skills);
-        }, function(error) {
+        _updateCollection, function(error) {
           alertsService.addWarning(
-            error || 'There was an error loading the collection.');
+            error || 'There was an error when loading the collection.');
         });
 
+    UndoRedoService.setOnChangedCallback(function(changeObject, wasApplied) {
+      if (changeObject && wasApplied &&
+          CollectionUpdateService.isAddingCollectionNode(changeObject)) {
+        var explorationId = (
+          CollectionUpdateService.getExplorationIdFromChangeObject(
+            changeObject));
+        var collectionNode = (
+          $scope.collection.getCollectionNodeByExplorationId(explorationId));
+        ExplorationSummaryBackendApiService
+          .loadPublicAndPrivateExplorationSummaries(
+          [explorationId]).then(function(summaries) {
+            var summaryBackendObject = {};
+            if (summaries.length != 0 && summaries[0].id == explorationId) {
+              summaryBackendObject = summaries[0];
+              summaryBackendObject.exists = true;
+            } else {
+              summaryBackendObject.exists = false;
+            }
+            collectionNode.setExplorationSummaryObject(summaryBackendObject);
+            _validateCollection();
+          }, function() {
+            alertsService.addWarning(
```

Comment:
Okay, this required a bit of thought. I moved the summary loading work to the detail editor. The change list will now only be modified for a new exploration node under two conditions:
1. The backend returned a result for the exploration summary (rather than requesting the summary afterwards)
2. The summary exists (a non-existent exploration will not even be added to the collection now)

I think the deferred execution here is fine after the changes. The user will not be able to interact with the exploration until the results return, so it's fine if other changes are added between them.

Also, if the user somehow tries adding the same exploration twice (and the first addition's summary request lags sufficiently that the trivial check for duplicate explorations fails), then it should fail silently in the collection object factory, which seems like the right behavior.